enhancements/cluster-logging/cluster-logging-move-init-to-es-operator.md

```diff
@@ -0,0 +1,73 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+reviewers:
+  - "@bparees"
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@alanconway"
+  - "@ewolinetz"
+approvers:
+  - "@ewolinetz"
+  - "@jcantrill"
+creation-date: 2019-09-17
+last-updated: 2019-10-25
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-es-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of moving Elasticsearch initialization logic to the elasticsearch-operator is to reduce the number of times these init scripts executed.
+At the moment we keep these scripts in the ES image, and execute them everytime when ES pod starts. 
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that allows users to specify init procedures for the ES cluster
+* Allow users to choose what command has to be executed during the init process for FULL and ROLLING elasticsearch cluster restart
+
+## Proposal
+
+The following changes will define how the elasticsearch-operator executes init process for Elasticsearch cluster restarts
+
+
+* Adding additional section to the elasticsearch-operator called `bootstrap`, which may look like this:
+
+    ```json
+    "bootstrap": {
+      "image": "quay.io/openshift/origin-logging-elasticsearch-bootstrap6:latest",
```

Comment:
What is the use case here to allow a custom image? Do we want to share control of which actions are packaged and executed on target clusters? Could users define here custom actions? If yes on latter, how can we know in future what is a canonical init procedure that we are able to support?

Furthermore, if users can provide custom images with actions for their clusters, how to resolve conflicts with our base init scripts, e.g. our are mandatory or conflict with user actions?

Reply:
The use case is this:  you mr administrator, spec your cluster along with an image and entrypoint for managing the cluster initialization during boot scenarios